### ISSUE TITLE
Nimbus is sending `exited` validators to MEV-Boost but should only send `active` or `pending` ones

### ISSUE THREAD

**Issue by jclapis** [status-im/nim-beacon-chain#3961]
I recently set up MEV-Boost for Goerli/Prater. I have an exited validator on Prater (https://prater.beaconcha.in/validator/0xb440a2621abcda9f2af03b31040d62cd5ea26a9aff16d09003c1cb566fc45da1b67924384375c460a0bf1b3c187ec9b1) and during Nimbus's validator registration process, it sends this validator over to MEV-Boost which causes it to fail.

According to [the official spec](https://ethereum.github.io/builder-specs/#/Builder/registerValidator), only `active` or `pending` validators should be sent to the builder during validator registration.

Here are MEV-Boost's logs:
```
mev-boost_1      | time="2022-08-12T00:08:00Z" level=info msg="http: GET /eth/v1/builder/status 200" duration=0.172181701 method=GET module=service path=/eth/v1/builder/status status=200
mev-boost_1      | time="2022-08-12T00:08:00Z" level=warning msg="error calling registerValidator on relay" error="HTTP error response: 400 / {\"code\":400,\"message\":\"not a known validator: 0xb440a2621abcda9f2af03b31040d62cd5ea26a9aff16d09003c1cb566fc45da1b67924384375c460a0bf1b3c187ec9b1\"}\n" method=registerValidator module=service numRegistrations=36 ua="nim-presto/0.0.3 (arm64/linux)" url="https://builder-relay-goerli.flashbots.net/eth/v1/builder/validators?id=rocketpool"
mev-boost_1      | time="2022-08-12T00:08:00Z" level=info msg="http: POST /eth/v1/builder/validators 502" duration=0.117795912 method=POST module=service path=/eth/v1/builder/validators status=502
```

Here are Nimbus's logs:
```
eth2_1           | INF 2022-08-12 04:43:12.000+00:00 Slot start                                 topics="beacnde" slot=3647616 epoch=113988 sync=synced peers=20 head=97449245:3647615 finalized=113985:0e76a2b3 delay=614us863ns
eth2_1           | WRN 2022-08-12 04:43:12.385+00:00 registerValidators: Couldn't register validator with MEV builder topics="beacval" registerValidatorResult="(status: 502, contentType: \"application/json\", data: @[123, 34, 99, 111, 100, 101, 34, 58, 53, 48, 50, 44, 34, 109, 101, 115, 115, 97, 103, 101, 34, 58, 34, 110, 111, 32, 115, 117, 99, 99, 101, 115, 115, 102, 117, 108, 32, 114, 101, 108, 97, 121, 32, 114, 101, 115, 112, 111, 110, 115, 101, 34, 125, 10])"
```

I [reported this to Flashbots](https://github.com/flashbots/mev-boost/issues/245), who pointed me to the builder API spec above. 

This issue is also present in [Lighthouse](https://github.com/sigp/lighthouse/issues/3465) and [Teku](https://github.com/ConsenSys/teku/issues/6070).

**To Reproduce**
1. Platform details (OS, architecture): Any
2. Branch/commit used: https://github.com/status-im/nimbus-eth2/commit/3793103
3. Commands being executed: Typical BN operation
4. Relevant log lines: See above

- Exit a validator
- Start Nimbus, which will load the keys for the exited validator
- Connect to MEV-boost as the block builder
- Observe the above behavior

**Comment by jclapis** [comment]
The issue is here: https://github.com/status-im/nimbus-eth2/blob/052f9edfd49a06bac5b625d8f792abed7ed2d5e7/beacon_chain/validators/validator_duties.nim#L1228

This needs to filter the validators by status. I would offer a PR but my system isn't working with Nim properly right now, I'll see if I can fix that when I have some time.

**Comment by zah** [comment]
This issue has been addressed by the linked PR above and the fixes have been included in the Nimbus v22.8.0 release.
